compiler/rustc_parse/src/parser/item.rs

```diff
@@ -1725,7 +1725,16 @@ impl<'a> Parser<'a> {
                 err.help("see https://doc.rust-lang.org/book/ch05-03-method-syntax.html for more information");
                 err
             } else {
-                self.expected_ident_found()
+                let mut err = self.expected_ident_found();
+                if self.check_ident() {
+                    self.bump();
+                    let (ident, _) = self.ident_or_err()?;
+                    err.help("the let keyword is not allowed in struct field definitions");
+                    err.help("see https://doc.rust-lang.org/book/ch05-01-defining-structs.html for more information");
+                    err.emit();
+                    return Ok(ident);
+                }
+                err
```

Comment:
I think it might be wrong because `ident_or_err` didn't `bump` (which is why there might be another `bump` below before returning `Ok`, so maybe you need to do yet another `bump` before returning here and that would fix things? The parser didn't advance as many tokens as you thought it had? I'll admit the current behavior is weird.

Reply:
Bumping before returning didn't seem to help, look at the latest commit CI logs when it finishes

Reply:
<FIXED>